### PR TITLE
MSSQL: Change regex to validate Provider connection string

### PR DESCRIPTION
**Pull request by ianselmi**
**What this PR does / why we need it**:
This PR allow specifie another conenction properties for MSSQL connection string. My need is specify ApplicationIntent property having instance with a port number, like this:
`127.0.0.1:123;ApplicationIntent:ReadOnly`
before this PR I could not specify ApplicationIntent with portname

Simple valid test:

- 127.0.0.1 -> valid
- 127.0.0.1:333 -> valid
- 127.0.0.1\instance -> valid
- 127.0.0.1\instance:233 -> valid
- 127.0.0.1\instance;ApplicationIntent=ReadOnly -> valid
- 127.0.0.1\instance:333;ApplicationIntent=ReadOnly -> valid
- 127.0.0.1\instance:ddd -> not valid
- 127.0.0.1\instance:ddd;ApplicationIntent=ReadOnly -> not valid

### CODE REVIEW

**Review by achatterjee-grafana — COMMENTED** [review]
Need dev review.

**Comment by ianselmi** [comment]
Do I need to do anything?

**Comment by kirederik** [comment]
Hey @ianselmi ,

Thanks for the PR and sorry for the delay in getting back to you. Would you mind adding a test case for the change?

Thanks!

**Comment by ianselmi** [comment]
> Hey @ianselmi ,
> 
> Thanks for the PR and sorry for the delay in getting back to you. Would you mind adding a test case for the change?
> 
> Thanks!

Hi @kirederik,
I added some tests for this PR

File: pkg/tsdb/mssql/mssql_test.go [hunk 1]
```diff
@@ -1330,6 +1330,46 @@ func TestGenerateConnectionString(t *testing.T) {
 				JsonData: sqleng.JsonData{},
 			},
 			expConnStr: "server=localhost;database=database;user id=user;password=;",
+		},
```
Comment (zoltanbedi) [inline pkg/tsdb/mssql/mssql_test.go:1333]:
```suggestion
		},
```

File: docs/sources/datasources/mssql.md [hunk 2]
```diff
@@ -14,19 +14,19 @@ Grafana ships with a built-in Microsoft SQL Server (MS SQL) data source plugin t
...
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Name`           | The data source name. This is how you refer to the data source in panels and queries.                                                                                                                                                                 |
+| `Default`        | Default data source means that it will be pre-selected for new panels.                                                                                                                                                                                |
+| `Host`           | The IP address/hostname and optional port of your MS SQL instance. If you omit the port, then the driver default is used (0). You can specify multiple connection properties such as ApplicationIntend using ';' character to separate each property. |
```
Comment (zoltanbedi) [inline docs/sources/datasources/mssql.md:21]:
```suggestion
| `Host`           | The IP address/hostname and optional port of your MS SQL instance. If you omit the port, then the driver default is used (0). You can specify multiple connection properties such as ApplicationIntent using ';' character to separate each property. |
```

**Review by zoltanbedi — COMMENTED** [review]
Hi @ianselmi,
Thanks for creating this. I found two minor things, other than that looks good to me.

File: docs/sources/datasources/mssql.md (hunk 2 above)
Comment (ianselmi) [reply docs/sources/datasources/mssql.md]: fixed

**Comment by ianselmi** [comment]
@zoltanbedi 
Should I do something?

**Review by zoltanbedi — APPROVED** [review]
@ianselmi no sorry. I think this is good to go now. Thank you for contributing!

**Pull request merged**